semeio/reporter/reporter.py

```diff
@@ -0,0 +1,29 @@
+from .serialize_numpy import serialize_numpy
+from .serialize_json import serialize_json
+
+
+class Reporter(object):
+    _serializers = [serialize_numpy, serialize_json]
+
+    def __init__(self):
+        pass
+
+    def report(self, key, val):
+        if not isinstance(key, str):
+            raise TypeError("Report key must be of type str")
+        if key == "" or "/" in key:
```

Comment:
It's not certain that the user will know what happened here, if some workflow author has a buggy workflow reporting with either of these keys. So maybe include in the exception what key was faulty.

Also, I really think we should constrain it to alphanumeric characters and some select others (`-_.`). Non-printable characters and a bunch of others are not going to bode well cometh the cloud.